test/kino_explorer/data_frame_cell_test.exs

```diff
@@ -0,0 +1,216 @@
+      %{
+        columns: %{"hour" => :integer, "team" => :string, "weekday" => :string},
+        variable: "teams"
+      }
+    ]
+
+    assert_broadcast_event(kino, "set_available_data", %{"data_options" => ^data_options})
+  end
+
+  describe "code generation" do
+    test "source for a data frame without operations" do
+      attrs = build_attrs(%{})
+
+      assert DataFrameCell.to_source(attrs) == """
+             people\
+             """
+    end
+
+    test "source for a data frame with sorting" do
+      attrs = build_attrs(%{"sorting" => [%{"direction" => "asc", "sort_by" => "name"}]})
+
+      assert DataFrameCell.to_source(attrs) == """
+             people |> Explorer.DataFrame.arrange(asc: name)\
+             """
+    end
+
+    test "source for a data frame with multiple sorting" do
+      attrs =
+        build_attrs(%{
+          "sorting" => [
+            %{"direction" => "asc", "sort_by" => "name"},
+            %{"direction" => "desc", "sort_by" => "id"}
+          ]
+        })
+
+      assert DataFrameCell.to_source(attrs) == """
+             people |> Explorer.DataFrame.arrange(asc: name, desc: id)\
+             """
+    end
+
+    test "source for a data frame with filtering" do
+      attrs =
+        build_attrs(%{
+          "filters" => [
+            %{"column" => "name", "filter" => "==", "type" => "string", "value" => "Ana"}
+          ]
+        })
+
+      assert DataFrameCell.to_source(attrs) == """
+             people |> Explorer.DataFrame.filter(name == "Ana")\
+             """
+    end
+
+    test "source for a data frame with multiple filtering" do
+      attrs =
+        build_attrs(%{
+          "filters" => [
+            %{"column" => "name", "filter" => "==", "type" => "string", "value" => "Ana"},
+            %{"column" => "id", "filter" => "<", "type" => "integer", "value" => "2"}
+          ]
+        })
+
+      assert DataFrameCell.to_source(attrs) == """
+             people |> Explorer.DataFrame.filter(name == "Ana") |> Explorer.DataFrame.filter(id < 2)\
+             """
+    end
+
+    test "sour for a data frame with columns with spaces" do
+      root = %{"data_frame" => "df", "export_to" => "new_df"}
+
+      operations = %{
+        "sorting" => [
+          %{"direction" => "asc", "sort_by" => "full name"},
+          %{"direction" => "desc", "sort_by" => "id"}
+        ],
+        "filters" => [
+          %{"column" => "full name", "filter" => "==", "type" => "string", "value" => "Ana"},
+          %{"column" => "id", "filter" => "<", "type" => "integer", "value" => "2"}
+        ]
+      }
+
+      attrs = build_attrs(root, operations)
+
+      assert DataFrameCell.to_source(attrs) == """
+             new_df =
+               df
+               |> Explorer.DataFrame.arrange(asc: col("full name"), desc: id)
+               |> Explorer.DataFrame.filter(col("full name") == "Ana")
+               |> Explorer.DataFrame.filter(id < 2)\
```

Comment:
@josevalim do we want to add `require` to each cell or rely on the user following the error and adding it prior to the cell?

Reply:
Nevermind, I see we already add it conditionally!